modules/library/main/src/main/java/org/geotools/filter/LikeFilterImpl.java

```diff
@@ -130,16 +130,28 @@ public String getSQL92LikePattern() throws IllegalArgumentException {
     }
 
     public void setWildCard(String wildCard) {
+        if (wildCard.isEmpty() || wildCard.length() > 1) {
```

Comment:
Sounds good. Out of curiosity, I've checked the schemas:

```xml
<xsd:complexType name="PropertyIsLikeType">
<xsd:complexContent>
<xsd:extension base="fes:ComparisonOpsType">
<xsd:sequence>
<xsd:element ref="fes:expression" minOccurs="2" maxOccurs="2"/>
</xsd:sequence>
<xsd:attribute name="wildCard" type="xsd:string" use="required"/>
<xsd:attribute name="singleChar" type="xsd:string" use="required"/>
<xsd:attribute name="escapeChar" type="xsd:string" use="required"/>
<xsd:attribute name="matchCase" type="xsd:boolean" use="optional" default="true"/>
</xsd:extension>
</xsd:complexContent>
</xsd:complexType>
```

Hum... no length restrictions.  However, [the spec](http://docs.opengeospatial.org/is/09-026r2/09-026r2.html#51) talks about character, not a string:

> A combination of regular characters, the wildCard character, the singleChar character, and the escapeChar character define the pattern.

Would the machinery have worked at all, before, with strings longer than one char?

Reply:
It would work but you could really break things by doing so.

Reply:
I don't think `FilterFunction_strMatches` needs any changes - it just calls `StaticGeometry.strMatches`

Reply:
Right, I did not see StaticGeometry was changed, sorry.

Reply:
It isn't really necessary but null checks on these three setters could be more helpful than an NPE.